### ISSUE TITLE
Cannot find module cypress-image-diff-js/dist/plugin

### ISSUE THREAD

**Issue by CaiqueCoelho** [uktrade/cypress-image-diff#194]
I'm trying to run the plugin for the first time and I encounter the following problem, could you help me, please?

Error:
```
Error: Cannot find module '/Users/caiquecoelho/Documents/Projects/openid_server_code/node_modules/cypress-image-diff-js/dist/plugin'
    at createEsmNotFoundErr (node:internal/modules/cjs/loader:1098:15)
    at finalizeEsmResolution (node:internal/modules/cjs/loader:1091:15)
    at resolveExports (node:internal/modules/cjs/loader:567:14)
    at Function.Module._findPath (node:internal/modules/cjs/loader:636:31)
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:1063:27)
    at Function.Module._resolveFilename.sharedData.moduleResolveFilenameHook.installedValue [as _resolveFilename] (/Users/caiquecoelho/Library/Caches/Cypress/13.6.0/Cypress.app/Contents/Resources/app/node_modules/@cspotcode/source-map-support/source-map-support.js:811:30)
    at Function.Module._load (node:internal/modules/cjs/loader:922:27)
    at Module.require (node:internal/modules/cjs/loader:1143:19)
    at require (node:internal/modules/cjs/helpers:119:18)
    at Object.<anonymous> (/Users/caiquecoelho/Documents/Projects/openid_server_code/cypress.config.ts:3:1)
    at Module._compile (node:internal/modules/cjs/loader:1256:14)
    at Module.m._compile (/Users/caiquecoelho/Library/Caches/Cypress/13.6.0/Cypress.app/Contents/Resources/app/node_modules/ts-node/dist/index.js:857:29)
    at Module._extensions..js (node:internal/modules/cjs/loader:1310:10)
    at Object.require.extensions.<computed> [as .ts] (/Users/caiquecoelho/Library/Caches/Cypress/13.6.0/Cypress.app/Contents/Resources/app/node_modules/ts-node/dist/index.js:859:16)
    at Module.load (node:internal/modules/cjs/loader:1119:32)
    at Function.Module._load (node:internal/modules/cjs/loader:960:12)
    at Module.require (node:internal/modules/cjs/loader:1143:19)
    at require (node:internal/modules/cjs/helpers:119:18)
    at loadFile (/Users/caiquecoelho/Library/Caches/Cypress/13.6.0/Cypress.app/Contents/Resources/app/packages/server/lib/plugins/child/run_require_async_child.js:89:14)
    at EventEmitter.<anonymous> (/Users/caiquecoelho/Library/Caches/Cypress/13.6.0/Cypress.app/Contents/Resources/app/packages/server/lib/plugins/child/run_require_async_child.js:116:38)
    at EventEmitter.emit (node:events:517:28)
    at EventEmitter.emit (node:domain:489:12)
```

cypress.config.ts
```
import { defineConfig } from 'cypress';
import getCompareSnapshotsPlugin from 'cypress-image-diff-js/plugin';

export default defineConfig({
  e2e: {
    setupNodeEvents(on, originalConfig) {
      const config = getCompareSnapshotsPlugin(on, originalConfig);
      require('@cypress/grep/src/plugin')(config);
      getCompareSnapshotsPlugin(on, config);
      return require('./cypress/plugins/index.js')(on, config);
    },
    experimentalRunAllSpecs: true,
    specPattern: 'cypress/e2e/**/*.{js,jsx,ts,tsx}',
    baseUrl: 'http://localhost:9000',
    retries: {
      runMode: 1,
      openMode: 0,
    },
    experimentalMemoryManagement: true,
    numTestsKeptInMemory: process.env.CI ? 0 : 10,
  },
  chromeWebSecurity: false,
  defaultCommandTimeout: 10000,
  requestTimeout: 25000,
  responseTimeout: 25000,
  env: {
    grepFilterSpecs: true,
    grepOmitFiltered: true
  }
});

```

**System:**
Mac Os 14.1 (23B74)

**Cypress: 13.6.0**

**Comment by kien-ht** [comment]
@CaiqueCoelho We have just released version 2.1.0, please upgrade your package and see if the error still persists.

Side note: I've noticed you have initialized our plugin twice, can you confirm if it's just a copy and paste problem?

**Comment by camhost01** [comment]
Hi @kien-ht I recently updated the plugin and also read the previous issue. I followed the steps but still got an error.
Configuration
// cypress.config.js
const { defineConfig } = require("cypress");
const getCompareSnapshotsPlugin = require('cypress-image-diff-js/plugin');
module.exports = defineConfig({
  e2e: {
    video: false,
    setupNodeEvents(on, config) {
      
      return getCompareSnapshotsPlugin(on, config);
    },
  },
});
// cypress-image-diff.config.cjs
const config = {
    ROOT_DIR: 'cypress/Visua_Test',
    REPORT_DIR: 'cypress/Report/html-report',
    SCREENSHOTS_DIR: 'screenshots',
    FAILURE_THRESHOLD: 0.1,
    RETRY_OPTIONS: {
        log: true,
        limit: 2, // max number of iterations
        timeout: 30000, // time limit in ms
        delay: 300, // delay before next iteration, ms
      },
    FAIL_ON_MISSING_BASELINE: true,
    COMPARISON_OPTIONS: { threshold: 0.2 },
    JSON_REPORT: { 
        FILENAME: 'cypress_visual_report',
        OVERWRITE: false,
      }, 
    CYPRESS_SCREENSHOT_OPTIONS:{
        capture: 'fullPage',
        disableTimersAndAnimations: true,
    }
  };
  module.exports = config;

e2e.js
const compareSnapshotCommand = require('cypress-image-diff-js/command');
compareSnapshotCommand();
import './commands'

package-lock
<img width="516" alt="image" src="https://github.com/uktrade/cypress-image-diff/assets/39304271/f98d3333-906f-4327-9830-1397286b1e31">

**Comment by kien-ht** [comment]
@camhost01, I've just created a working integration example [here](https://github.com/kien-ht/cypress-image-diff-html-report/tree/16b18baab535e7dafd96453f779739905cba43f2/examples/javascript). Please take a look and see if you have any misconfig parts.

**Comment by camhost01** [comment]
Hi @kien-ht  I checked the **Java** configuration and they are the same. I notice that in your example, the cypress.config.js is the same compared with TypeScript. Is that correct? the issue that I facing is related in the const definition

<img width="1710" alt="image" src="https://github.com/uktrade/cypress-image-diff/assets/39304271/3b5709cf-ef0a-416b-abcb-6daaf7ab4be2">

Look what I have
cypress-image-diff.config.cjs
<img width="474" alt="image" src="https://github.com/uktrade/cypress-image-diff/assets/39304271/895faad4-2da8-451e-80cf-b07b3eef2f51">

cypress.config.js
<img width="692" alt="image" src="https://github.com/uktrade/cypress-image-diff/assets/39304271/e644eeb3-ef16-4d84-a29e-644dcb903401">

e2e.js
<img width="720" alt="image" src="https://github.com/uktrade/cypress-image-diff/assets/39304271/d5637f85-9431-429d-9c8d-c71a23db331b">

dependencies
<img width="471" alt="image" src="https://github.com/uktrade/cypress-image-diff/assets/39304271/7fea5d18-b708-4c0a-b61d-aa850e26bfa5">

I think I do not need an extra configuration to run the plugin, right?

**Comment by kien-ht** [comment]
Yes correct. Configuration file is optional, used in case you want to change the default behaviour of this plugin.

May I suggest you two things:
1. Clean cache the node_modules and reinstall again.
2. Pull the actual code of the example I provided above, and run it locally to see if you see any problems. I believe I wrote a README.md in the respective example folder, in case you struggle to make it run.

**Comment by Strezzo06** [comment]
> Hi @kien-ht I checked the **Java** configuration and they are the same. I notice that in your example, the cypress.config.js is the same compared with TypeScript. Is that correct? the issue that I facing is related in the const definition
> 
> <img alt="image" width="1710" src="https://private-user-images.githubusercontent.com/39304271/291100986-3b5709cf-ef0a-416b-abcb-6daaf7ab4be2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI4ODQwNTQsIm5iZiI6MTcwMjg4Mzc1NCwicGF0aCI6Ii8zOTMwNDI3MS8yOTExMDA5ODYtM2I1NzA5Y2YtZWYwYS00MTZiLWFiY2ItNmRhYWY3YWI0YmUyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjE4VDA3MTU1NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM3ODMwMWRiNTBiMjgzNzAyMWUxNzExODIzMzQyM2E5MTM4ZTk3ZWUzODNjZmUzNTA0MWFlNzMyNWMzN2QxZDgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.aWut-VrpFXZ-BOoJUZx9XjWMcuSOA0FXAWfNEW68l8k">
> Look what I have cypress-image-diff.config.cjs <img alt="image" width="474" src="https://private-user-images.githubusercontent.com/39304271/291101723-895faad4-2da8-451e-80cf-b07b3eef2f51.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI4ODQwNTQsIm5iZiI6MTcwMjg4Mzc1NCwicGF0aCI6Ii8zOTMwNDI3MS8yOTExMDE3MjMtODk1ZmFhZDQtMmRhOC00NTFlLTgwY2YtYjA3YjNlZWYyZjUxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjE4VDA3MTU1NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFmNjRlZGM5NWNkNWY4ODc2ZWY0NTBkNzE3YzMxNmI0ZjhhZGFmNWI0ZmUwYjdmZmI1NDQ1YmUyMmViNGU3OGImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.bUIl60GjVc7FDZ095PqeyFv4Tddl_WYwckPaHhGFHlQ">
> 
> cypress.config.js <img alt="image" width="692" src="https://private-user-images.githubusercontent.com/39304271/291101849-e644eeb3-ef16-4d84-a29e-644dcb903401.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI4ODQwNTQsIm5iZiI6MTcwMjg4Mzc1NCwicGF0aCI6Ii8zOTMwNDI3MS8yOTExMDE4NDktZTY0NGVlYjMtZWYxNi00ZDg0LWEyOWUtNjQ0ZGNiOTAzNDAxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjE4VDA3MTU1NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU3ODhiOTYxYzViMWRlYTY2MTA1MGExNWFmZWY1NjEwMmE0M2I0N2NjN2IxNzZhMTlkYWZmZDAxZWI1YmJjMTgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.3b4cpnPFKZWUEO3ITWtDMgaC287B6MdVVV0td2fCMFM">
> 
> e2e.js <img alt="image" width="720" src="https://private-user-images.githubusercontent.com/39304271/291102018-d5637f85-9431-429d-9c8d-c71a23db331b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI4ODQwNTQsIm5iZiI6MTcwMjg4Mzc1NCwicGF0aCI6Ii8zOTMwNDI3MS8yOTExMDIwMTgtZDU2MzdmODUtOTQzMS00MjlkLTljOGQtYzcxYTIzZGIzMzFiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjE4VDA3MTU1NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTlmMjRhYzM1NzcwOTBmOWYyMzA1ZjNjYjhmMWE1OGNjYmM4MzE5ZTlkYjFiNDU4ZDdkMWFkYjgzNzY1NTQxY2QmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.O9RA6WiccW0pNZm0szWFHZC0_bVzqVlep7gH82gkog8">
> 
> dependencies <img alt="image" width="471" src="https://private-user-images.githubusercontent.com/39304271/291102083-7fea5d18-b708-4c0a-b61d-aa850e26bfa5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI4ODQwNTQsIm5iZiI6MTcwMjg4Mzc1NCwicGF0aCI6Ii8zOTMwNDI3MS8yOTExMDIwODMtN2ZlYTVkMTgtYjcwOC00YzBhLWI2MWQtYWE4NTBlMjZiZmE1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjE4VDA3MTU1NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWUzZjNhYWM0MjQ3ODZlYTg3ZTNkMGRiODlmYjExNDRkNjQ4M2QzMDYyMjRmOGQyYjU1YjM4NTY3M2RlOWEzNzkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.pT9RrnLQoKcnQomBlt3SeLibnOLCUe_M8RctSM7mOfg">
> 
> I think I do not need an extra configuration to run the plugin, right?

I updated to V2.1.1 and did follow this step but still found error same as you
I tried to clean node_modules many times but still found this error
<img width="838" alt="Screen Shot 2023-12-18 at 2 35 35 PM" src="https://github.com/uktrade/cypress-image-diff/assets/60592070/e1f357f0-d8cf-4bec-9542-c5c5839267a2">

**Comment by dominikzogg** [comment]
Same error, using it in a NextJS Typescript project.

**Comment by Edwardmunn** [comment]
> @camhost01, I've just created a working integration example [here](https://github.com/kien-ht/cypress-image-diff-html-report/tree/main/examples). Please take a look and see if you have any misconfig parts.

I am seeing the same issue as others within this thread.
I have downloaded the example project linked, and ran the following: `npm install` and `npm run test`. I am presented with the following error:

![Screenshot 2023-12-18 at 11 28 36](https://github.com/uktrade/cypress-image-diff/assets/40793125/86f796a1-7c18-4931-81d8-99c5d3b2780d)

**Comment by Strezzo06** [comment]
> 🎉 This issue has been resolved in version 2.1.2 🎉
> 
> The release is available on:
> 
> * [npm package (@latest dist-tag)](https://www.npmjs.com/package/cypress-image-diff-js/v/2.1.2)
> * [GitHub release](https://github.com/uktrade/cypress-image-diff/releases/tag/v2.1.2)
> 
> Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot 📦🚀

Thank you for the excellent support. I was able to run through that point. However, I encountered another error. #198

**Comment by CaiqueCoelho** [comment]
Thank you so much @kien-ht  @PippoRaimondiDIT now it works :D
